src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -157,6 +156,31 @@ public void ProcessStart_UseShellExecute_OnUnix_SuccessWhenProgramInstalled(bool
             }
         }
 
+        [Fact]
+        [PlatformSpecific(~TestPlatforms.OSX)] // On OSX, ProcessName returns the script interpreter.
+        public void ProcessNameMatchesScriptName()
+        {
+            string scriptName = GetTestFileName();
+            string filename = Path.Combine(TestDirectory, scriptName);
+            File.WriteAllText(filename, $"#!/bin/sh\nsleep 600\n"); // sleep 10 min.
+            // set x-bit
+            int mode = Convert.ToInt32("744", 8);
+            Assert.Equal(0, chmod(filename, mode));
+
+            using (var process = Process.Start(new ProcessStartInfo { FileName = filename }))
+            {
+                try
+                {
+                    Assert.Equal(scriptName, process.ProcessName);
```

Comment:
should this actually return interpreter and script name in the arguments?

Reply:
It's debatable. I think the main use-case for this property is to help identify processes, then the script name is more useful. Also, when starting a process, scripts can be started the same way as native executables, so it may be unexpected something is actually a script and causes the interpreter to show up here.

Reply:
note: this has been returning the script name on Linux so far. I'm adding a test here because that regressed in https://github.com/dotnet/corefx/pull/37144 (`xdg-open` is a script).

Reply:
Honestly I'd never expect this to return anything else than interpreter name - I think we should match what `ps aux` does (if it does match then I'm fine with this change), otherwise we're just asking developers to implement their own interop layer to do the right thing. I'd rather add something like `ScriptName` instead.

Reply:
`ps aux` returns the full command line, which includes both the interpreter and the script name.
gnome system monitor (something like Windows process explorer) has a 'Process Name' column which shows the script names.

The `ProcessName` is the property that helps a user identify a certain process. Having a bunch of `bash` and `python` processes is not that helpful (same for `Process.GetProcessesByName`).

Reply:
@tmds I'm thinking from security point of view you could name your script something like `sleep` (no extension) and run it with bash - if you implement it this way the user will see `sleep` (possibly with path). I'd rather return `bash <scriptName>`

Reply:
@stephentoub @wtgodbe @danmosemsft can you chime in on what `ProcessName` should return for scripts?

Reply:
Is Windows a bad analog? It presumably returns cmd.exe.

Reply:
> It presumably returns cmd.exe.

I assume so too.

> Is Windows a bad analog?

Unix scripts are more closer to executables than Windows batch files. They are executables (+x bit set) and can be started directly from the Process class (no UseShellExecute required). For a user, there is no difference between a script 'executable' and a native executable. I think it will be unexpected and undesired that the interpreter shows up as the ProcessName.

Reply:
> Unix scripts are more closer to executables than Windows batch files. 

Fair enough. I don't feel strongly either way. @stephentoub thoughts?

Reply:
What do other languages / frameworks do here from their equivalent types, e.g. Java, node.js, Go, Rust, etc.?

Reply:
java's ProcessHandle provides the following:
- arguments: array of strings (from cmdline)
- command: string (from exe)
- commandLine: string (from cmdline)

rust's sysinfo Process provides:
- name: argv[0] (from cmdline)
- cmd: string (from cmdline)

I don't think there is a standard way of getting this info in node.js or go. It is suggested to read stdout from `ps ...`, which includes cmdline.

imo we shouldn't change ProcessName unless additional properties are added to Process that make it possible to identify a certain script (and even then we may want to keep the current behavior).